Feat/getter function for view key

Roee-87 opened the pull request:
<!--
    Thank you for submitting the PR! We appreciate you spending the time to work on these changes.

    Help us understand your motivation by explaining why you decided to make this change.

    Happy contributing!
-->

## Motivation

This PR closes [Issue 995](https://github.com/ProvableHQ/sdk/issues/995).  A method `tvk()` is added to the impl block for the `Transition` struct that returns a transition view key.  This method takes a reference to a `ViewKey` as an input.

## Test Plan

<!--
    If you changed any code,
    please provide us with clear instructions on how you verified your changes work.
    Bonus points for screenshots and videos!
-->

I added an `assert_eq` to the `test_transition_helpers` test function which checks that the newly added `tvk()` method returns the correct transition view key. (NOTE:  Placeholder string used instead of the calculated value.  Once I know how to manually compute the transition view key, I'll mark this PR as ready for review.)

Roee-87 committed:
added tvk method to Transition struct

Roee-87 committed:
added tvk test to test_transition_helpers()

Roee-87 committed:
fixed clippy error

Roee-87 committed:
ran cargo fmt

iamalwaysuncomfortable commented on wasm/src/ledger/transition.rs:
```diff
@@ -274,6 +285,11 @@ mod tests {
         );
+        assert_eq!(
+            transition.tvk(&view_key),
+            Field::from_string("_placeholder_") // Placeholder for the view key, will replace with manually computed value.
```
You can make this value the actual tvk, we tend to not work with any non-valid values in the test unless we're specifically trying to fail the test.

Roee-87 replied:
Done:  [7a9aab9](https://github.com/ProvableHQ/sdk/pull/996/commits/7a9aab93013d69cbe01173937efd028c6850d147)

iamalwaysuncomfortable reviewed (commented):
This looks good minus the test, once that's updated, this can be merged.

Roee-87 committed:
added test private key and value for tvk

Roee-87 committed:
Merge branch 'mainnet' into feat/getter-function-for-view-key

Roee-87 committed:
ran cargo fmt

Roee-87 committed:
ran cargo fmt

Roee-87 committed:
added back line 14 of package.json

For some reason, local testing removed the "type": "module" field in the package.json.

Signed-off-by: Roy Rotstein <112157037+Roee-87@users.noreply.github.com>

Roee-87 committed:
reinsterted original package.json

Roee-87 committed:
replaced Cargo.lock and package.json with versions on main

iamalwaysuncomfortable commented on wasm/src/ledger/transition.rs:
```diff
@@ -176,6 +176,13 @@ impl Transition {
+    /// Get the transition view key of the transition.
+    pub fn tvk(&self, view_key: &ViewKey) -> Field {
+        let tpk = self.tpk();
+        let tvk = tpk.scalar_multiply(&view_key.to_scalar()).to_x_coordinate();
+        tvk
```
We should make it a bit more concise
```suggestion
        tpk.scalar_multiply(&view_key.to_scalar()).to_x_coordinate()
```

iamalwaysuncomfortable reviewed (commented):
One minor suggestion and then this is good to go.

Roee-87 committed:
Update wasm/src/ledger/transition.rs

Co-authored-by: Mike Turner <mike@provable.com>
Signed-off-by: Roy Rotstein <112157037+Roee-87@users.noreply.github.com>
